app/src/androidTest/java/com/github/se/polyfit/viewmodel/MealViewModelTest.kt

```diff
@@ -0,0 +1,224 @@
+package com.github.se.polyfit.viewmodel
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.MutableLiveData
+import com.github.se.polyfit.data.local.dao.MealDao
+import com.github.se.polyfit.data.remote.firebase.MealFirebaseRepository
+import com.github.se.polyfit.data.repository.MealRepository
+import com.github.se.polyfit.model.ingredient.Ingredient
+import com.github.se.polyfit.model.meal.Meal
+import com.github.se.polyfit.model.meal.MealOccasion
+import com.github.se.polyfit.model.nutritionalInformation.MeasurementUnit
+import com.github.se.polyfit.model.nutritionalInformation.Nutrient
+import com.github.se.polyfit.model.nutritionalInformation.NutritionalInformation
+import com.github.se.polyfit.viewmodel.meal.MealViewModel
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.mockk
+import kotlin.test.assertFailsWith
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.runBlockingTest
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class MealViewModelTest {
+  @get:Rule val rule = InstantTaskExecutorRule()
+
+  private val mealRepo: MealRepository = mockk()
+  private lateinit var viewModel: MealViewModel
+
+  @Before
+  fun setUp() {
+    Dispatchers.setMain(TestCoroutineDispatcher())
+
+    val mealDao = mockk<MealDao>()
+    val mealFirebaseRepository = mockk<MealFirebaseRepository>()
+
+    val meal =
+        Meal(
+            occasion = MealOccasion.OTHER,
+            name = "Test Meal",
+            mealID = 1,
+            nutritionalInformation = NutritionalInformation(mutableListOf()))
+
+    every { mealDao.getMealByFirebaseID(any()) } returns meal
+    coEvery { mealRepo.getMeal(any()) } returns meal
+
+    viewModel = MealViewModel(mealRepo)
+    viewModel.setMealData(meal)
+    viewModel.setFirebaseID("firebase123")
+  }
+
+  @After
+  fun tearDown() {
+    // Clear all MockK confirmations and stubbings
+    io.mockk.clearMocks(mealRepo)
+  }
+
+  @Test
+  fun testInitialization_withFirebaseID_loadsData() {
+    assert(viewModel.meal.value?.name == "Test Meal")
+  }
+
+  @Test
+  fun testSetMealName_updatesMealName() {
```

Comment:
this test will no longer be needed if you remove `setMealName` and use `setMealData` only